src/app/static/src/tests/mocks.ts

```diff
@@ -47,7 +60,7 @@ export const mockGitState = (props: RecursivePartial<GitState> = {}): GitState =
             changelog_types: ["published", "internal"],
             ...props.metadata
         } : null
-    }
+    } as GitState // Would be better to alter the recusive partial type above rather than type casting
```

Comment:
This is not ideal, so if anyone has any advice here, let me know

Reply:
Why does it need this cast at all? If every property from `GitState` is guaranteed to exist, it will be of type `GitState` - which properties are possibly missing?

Reply:
I get this type error: x-special/nautilus-clipboard
copy
recent:///5bb2fa9bf2624e8343d4189c62715491

Reply:
The recursive type for reference:
![image](https://user-images.githubusercontent.com/57633087/166493764-14ed8f4d-5d7f-408b-9b98-012e42baace1.png)

Reply:
Ok, the problem is because the recursive partial allows you to pass incomplete `GitCommit` objects. Solution:

```
export const mockGitState = (props: RecursivePartial<GitState> = {}): GitState => {
    return {
        branches: ["master", "dev"],
        selectedBranch: "",
        selectedCommit: "",
        ...props,
        commits: props.commits !== null ? props.commits.map((c) => mockCommit(c)) : [],
        metadata: props.metadata !== null ? {
            instances_supported: false,
            git_supported: true,
            instances: {"source": []},
            changelog_types: ["published", "internal"],
            ...props.metadata
        } : null
    }
}
```

Reply:
Thanks Alex, I wouldn't have thought to do that